client/src/main/java/org/eclipse/hono/client/impl/CommandClientImpl.java

```diff
@@ -186,10 +188,19 @@ protected BufferResult getResult(final int status, final String contentType, fin
             MessageHelper.setPayload(request, contentType, data);
             sendRequest(request, responseTracker.completer(), null, currentSpan);
 
-            return responseTracker.map(ignore -> null);
+            return responseTracker.recover(t -> {
```

Comment:
Why don't we set the status code in `AbstractRequestResponseClient`'s *sendRequest* method? That way we would need to handle status code in a single place only ...

Reply:
Yes, I've amended the commit accordingly.